native/src/main/java/io/ballerina/stdlib/grpc/protobuf/cmd/GrpcCmd.java

```diff
@@ -118,14 +119,56 @@ public void execute() {
             return;
         }
 
-        // check input protobuf file path
-        Optional<String> pathExtension = getFileExtension(protoPath);
-        if (pathExtension.isEmpty() ||
-                !PROTO_EXTENSION.equalsIgnoreCase(pathExtension.get())) {
-            String errorMessage = "Invalid proto file path. Please input valid proto file location.";
-            outStream.println(errorMessage);
-            return;
+        File input = new File(protoPath);
+        if (input.isDirectory()) {
+            // Multiple proto files
+            List<String> protoFiles;
+            try {
+                protoFiles = getProtoFiles(Paths.get(input.getAbsolutePath()));
+            } catch (IOException e) {
+                String errorMessage = "Failed to find proto files in the directory. " +
+                        "Please input a valid proto files directory.";
+                outStream.println(errorMessage);
+                return;
+            }
+            if (protoFiles.size() == 0) {
+                String errorMessage = "Input directory does not contain any proto files. " +
+                        "Please input a valid proto files directory.";
+                outStream.println(errorMessage);
+            } else {
+                for (String protoFile : protoFiles) {
+                    generateBalFile(protoFile);
+                }
+            }
+        } else {
+            // Single proto file
+            // check input protobuf file path
+            Optional<String> pathExtension = getFileExtension(protoPath);
+            if (!PROTO_EXTENSION.equalsIgnoreCase(pathExtension.get())) {
+                String errorMessage = "Invalid proto file path. Please input valid proto file location.";
+                outStream.println(errorMessage);
+                return;
+            }
+            generateBalFile(protoPath);
+        }
+    }
+
+    private List<String> getProtoFiles(Path path) throws IOException {
+        if (!Files.isDirectory(path)) {
+            throw new IllegalArgumentException("Path must be a directory!");
         }
```

Comment:
Do we need to check this here. as we already check before

Reply:
Removed in 714b52c